bindings/java/src/main/com/apple/foundationdb/tuple/ByteArrayUtil.java

```diff
@@ -45,48 +46,56 @@
 	 *  concatenated elements.
 	 * @param parts the pieces to be joined. May be {@code null}, but does not allow
 	 *  for elements in the list to be {@code null}.
-	 *
+	 * 
 	 * @return a newly created concatenation of the input
 	 */
 	public static byte[] join(byte[] interlude, List<byte[]> parts) {
+		return interludeJoin(interlude, parts.toArray(new byte[0][]));
+	}
+	/**
+	 * Joins a set of byte arrays into a larger array. The {@code interlude} is placed
+	 *  between each of the elements, but not at the beginning or end. In the case that
+	 *  the list is empty or {@code null}, a zero-length byte array will be returned.
+	 *
+	 * @param interlude can be {@code null} or zero length. Placed internally between
+	 *  concatenated elements.
+	 * @param parts the pieces to be joined. May be {@code null}, but does not allow
+	 *  for elements in the array to be {@code null}.
+	 *
+	 * @return a newly created concatenation of the input
+	 */
+	public static byte[] interludeJoin(byte[] interlude, byte[][] parts) {
 		if(parts == null)
-			return new byte[0];
-		int partCount = parts.size();
-		if(partCount == 0)
-			return new byte[0];
-
-		if(interlude == null)
-			interlude = new byte[0];
-
-		int elementTotals = 0;
-		int interludeSize = interlude.length;
-		for(byte[] e : parts) {
-			elementTotals += e.length;
+		return new byte[0];
```

Comment:
It looks like everything in this function starting at this line is indented one tab too far to the left.

Reply:
@ajbeamon Can you take another look?